/*
  Vehicle Data

  This API section holds messages related to OTA Vehicle Data between device and backend.

  The version of the OpenAPI document: 3

  Generated by OpenAPI Generator: https://openapi-generator.tech
*/

syntax = "proto3";

package com.bosch.bfb.protobuf.vehicledata.v3;

import public "device_to_backend_message_meta.proto";

message MeasurementResultEvent {

  DeviceToBackendMessageMeta meta = 3347973;

  // URI of the measurement result, must be set if no measurementResult is set.  {scheme}://{path} The scheme of the URI identifies the technology of the object storage. The path identifies  the measurement result relative to the location. If the scheme contains a basic technology/protocol  such as http(s), the receiver of this event interprets the artifactUri as an ordinary URL.  Examples: * s3://{bucketName}/{groupName}/{updateArtifact} * bosch-cs://blobs/{contentId}
  string artifactUri = 56320456;

  // Serialization method of measurementConfiguration, e.g. base64
  string serialization = 385936369;

  // Serialized measurement result, must be set if no artifactUri is set. 
  string measurementResult = 169291017;

  // Information about the stack used e.g. AA-Stack
  string stackType = 117326309;

  // Start of the measurement data in this message
  string executionStart = 407767000;

  // End of the measurement data in this message
  string executionEnd = 143454559;

  // The id of the measurementConfiguration for this data
  string measurementId = 44754426;

  // The version of the measurementConfiguration for this data
  string jobVersion = 327820711;

  // The vin of the vehicle this device was part of when this measurement data has been collected
  string vin = 116763;

  // Is this the last measurement result for this measurementInstance (activityId)
  string isLast = 106598690;

}
